### PR TITLE
[REF] web_tour: replace _legacyIsVisible by hoot isVisible

### PR DESCRIPTION
**Pull request by pipu-odoo**
In this commit, we replace the use of _legacyIsVisible by the visible option of hoot which refers to isVisible function of hoot-dom.js. This change implies changes in a few tours.
For information, _legacyIsVisible accepts elements with a width of 0 or a length of 0 (which is a bit absurd) while isVisible does not, which implies the changes where we just add :not(:visible) to the trigger. The other main modifications implies tours where you have to hover the trigger to then see the element you want to click on.

task~3974087

https://github.com/odoo/enterprise/pull/68080

### CODE REVIEW

File: addons/hr_skills/static/tests/tours/skills_tour.js
Comment (Bertrand2) [inline addons/hr_skills/static/tests/tours/skills_tour.js:1]: Would it be possible to separate the formatting in a separate commit? It makes the actual changes not that visible. Moreover, I see that some strings are not formatted correctly (`'text "text" text'` instead of `"text 'text' text"`), those could be changed as well :)

File: addons/web_tour/static/src/tour_service/tour_step_automatic.js [hunk 1]
```diff
@@ -188,15 +188,22 @@ export class TourStepAutomatic extends TourStep {
      * @returns {HTMLElement}
      */
     findTrigger() {
-        let nodes;
+        let triggerEl;
         try {
-            nodes = hoot.queryAll(this.trigger);
+            //TODO: this condition has to be removed after hoot bug has been fixed
```
Comment (pipu-odoo) [inline addons/web_tour/static/src/tour_service/tour_step_automatic.js:193]: @Arcasias 
https://media.discordapp.net/attachments/1096043239250853958/1269937553394368583/image.png?ex=66b72703&is=66b5d583&hm=d219b414ecd2ae191d56f8037b09ba4a46b0ed991fdc33ddf97d55598d3eff0b&=&format=webp&quality=lossless&width=607&height=468

**Review by Arcasias — APPROVED** [review]
@robodoo r+

**Comment by Iucapad** [comment]
@robodoo r+

**Comment by Iucapad** [comment]
@robodoo rebase-ff r+

File: addons/website/static/tests/tours/drop_404_ir_attachment_url.js [hunk 2]
```diff
@@ -1,53 +1,74 @@
...
-            const imgEl = this.anchor;
-            if (!imgEl.complete
-                || imgEl.naturalWidth !== 0
-                || imgEl.naturalHeight !== 0) {
-                throw new Error('This is supposed to be a 404 image');
-            }
+    () => [
+        {
+            trigger: ".o_website_preview.editor_enable.editor_has_snippets",
+            noPrepend: true,
```
Comment (qsm-odoo) [inline addons/website/static/tests/tours/drop_404_ir_attachment_url.js:19]: Is this a new set convention about how to indent tour files? I would like that (well I would like any choice because I am never sure how we want to indent those), but maybe it would be nice to make an unique commit indenting them all the right way? Anyways, detail but this makes it less easy to see the change you are asking me to check, and hides the critical part of your commit, which seems important.

Comment (qsm-odoo) [inline addons/website/static/tests/tours/drop_404_ir_attachment_url.js:19]: Is not this something that is automatically done by `registerWebsitePreviewTour`?

File: addons/website/static/tests/tours/drop_404_ir_attachment_url.js [hunk 3]
```diff
@@ -1,53 +1,74 @@
...
-    url: '/',
-    edition: true,
-}, () => [
-    ...dragNDrop({
-        id: 's_404_snippet',
-        name: '404 Snippet',
-        groupName: "Images",
-    }),
+registerWebsitePreviewTour(
+    "drop_404_ir_attachment_url",
```
Comment (qsm-odoo) [inline addons/website/static/tests/tours/drop_404_ir_attachment_url.js:10]: I don't understand this whole tour change actually, is it still testing the same thing? Why is it dragging another unrelated snippet at the start, etc? Maybe needs to be into its own commit?

File: addons/website/static/tests/tours/website_form_editor.js [hunk 4]
```diff
@@ -952,9 +952,6 @@ registerWebsitePreviewTour("website_form_editable_content", {
         trigger: ":iframe .o_overlay_move_options .o_move_handle",
         run: "drag_and_drop :iframe section.s_website_form",
     },
-    {
-        trigger: ":iframe section.s_website_form .col-lg-4[contenteditable=true]",
-    },
```
Comment (qsm-odoo) [inline addons/website/static/tests/tours/website_form_editor.js:957]: I think it is a mistake, if I remember it well we put that recently to fix a race condition. It definitely needed a comment though, strange we merged it that way. Something like
```
// TODO we should ensure that as soon as the snippet is visible, it is editable to avoid this step
```
In any case, just keep this step please :pray:

File: addons/website/static/tests/tours/website_form_editor.js [hunk 5]
```diff
@@ -970,8 +967,10 @@ registerWebsitePreviewTour("website_form_editable_content", {
         content: "Check that the new text value was correctly set",
         trigger: ":iframe section.s_website_form h3:contains(/^ABC$/)",
     },
-    {   content: "Remove the dropped column",
-        trigger: ":iframe .oe_overlay.oe_active .oe_snippet_remove",
+    {
+        content: "Remove the dropped column",
+        //To remove :not(:visible) of the trigger, we should scrollTo top of body.
+        trigger: ":iframe .oe_overlay.oe_active .oe_snippet_remove:not(:visible)",
```
Comment (qsm-odoo) [inline addons/website/static/tests/tours/website_form_editor.js:973]: I understood your commit message's point but I don't get this one. Can't this be fixed the same way as other by first hovering the snippet ?

All in all, I guess you know it but relying on the presence of `:visible` in the trigger string seems a bit weird. If the sole use case if something that has no with or height, surely the tour is indeed wrong in the first place as you can't really expect users to interact with those and thus the tour should be changed somehow (by hovering first like here?)

**Review by qsm-odoo — CHANGES_REQUESTED** [review]
I only checked the `website` folder files, thanks for checking :pray:

File: addons/website/static/tests/tours/website_form_editor.js (hunk 5 above)
Comment (pipu-odoo) [reply addons/website/static/tests/tours/website_form_editor.js:973]: I actually tried to do the hover and click in this case, but it doesn't work. I'll add a TODO to do it later and not block this PR.

Concerning the use of :visible or rather :not(:visible) in the tours, it is indeed not "logical" that we find ourselves with invisible triggers. However, in some cases (as in enterprise/test_l10n_be_hr_payroll_account/static/tests/tours/hr_contract_salary_tour.js ), this feature is useful.
